kafka-client/src/main/java/dev/responsive/db/CassandraKeyValueSchema.java

```diff
@@ -95,10 +90,75 @@ public void create(final String tableName) {
     );
   }
 
+  /**
+   * Initializes the metadata entry for {@code table} by adding a
+   * row with key {@code _metadata} and sets special columns
+   * {@link ColumnNames#OFFSET} and {@link ColumnNames#EPOCH}.
+   *
+   * <p>Note that this method is idempotent as it uses Cassandra's
+   * {@code IF NOT EXISTS} functionality.
+   *
+   * @param table          the table that is initialized
+   * @param kafkaPartition the partition to initialize
+   */
   @Override
-  public void prepare(final String tableName) {
-    metadataStatements.prepare(tableName);
+  public FencingToken init(
```

Comment:
I moved initialization code all into this method - it both creates the metadata rows if they don't exist and reserves the offset, this sets it up for a future Schema implementation that doesn't use epochs or fencing tokens (returns NoOpFencingToken) at all, but does use offsets.